WalletWasabi/Wallets/FileSystemBlocksRepository.cs

```diff
@@ -0,0 +1,133 @@
+using NBitcoin;
+using NBitcoin.DataEncoders;
+using Nito.AsyncEx;
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Wallets
+{
+	/// <summary>
+	/// FileSystemBlocksRepository is a blocks repository that keep the blocks in the file system.
+	/// </summary>
+	public class FileSystemBlocksRepository : IRepository<uint256, Block>
```

Comment:
Maybe the name of the class should be `BlocksFileSystemRepository`.

Is the `s` in `Blocks` for plural or for possession?
If it is for plural then maybe `BlockFolderLock` should be `BlocksFolderLock`.
If it is possessive then maybe we should remove it altogether.

Reply:
The `s` here is for plural but I think I should use the singular word `Block` here because that's how we are naming things in the project. For example, the class that manages the keys is called KeyManager and not KeysManager (BlockFolder is another good example) so, I will use singular.